qemu/tests/file_copy_between_blk_scsi.py

```diff
@@ -0,0 +1,88 @@
+import logging
+from virttest import utils_misc
+from virttest import error_context
+from avocado.core import exceptions
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Copy file between block disk and scsi disk:
+    1) Boot guest with both block disk and scsi disk attached.
+    2) Get disk dev file name in guest
+    3) Format those disks in guest
+    4) Copy/dd file to disk0
+    5) Copy file from disk0 to disk1
+    6) Check md5sum of file after copy
+    7) Copy file from disk1 to disk0
+    8) Check md5sum of file after copy
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    def copy_files(session, src_disk, dst_disk):
+        """
+         Copy file from src_disk to dst_disk in guest
+        :param session: VM session
+        :param src_disk: the src disk
+        :param dst_disk: the dst disk
+        """
+        copy_cmd = params["copy_cmd"]
+        file_name = params["file_name"]
+        copy_timeout = int(params.get("copy_timeout", 360))
+
+        src_file = "".join([src_disk, file_name])
```

Comment:
Using '.join' here would induce a file name as '/vdatest' or '/sdbtest', while 'os.path.join' without the problem.

Reply:
Besides, for windows guest, there is "TypeError: sequence item 0: expected string, NoneType found" problem; and even using 'os.path.join', there is still "AttributeError: 'NoneType' object has no attribute 'endswith'" problem.